community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala

```diff
@@ -104,126 +104,142 @@ final case class SbtCommunityProject(project: String, sbtTestCommand: String,
       s"--addPluginSbtFile=$sbtPluginFilePath")
 
 object projects
-  val utest = MillCommunityProject(
+  lazy val utest = MillCommunityProject(
     project = "utest",
     baseCommand = s"utest.jvm[$compilerVersion]",
   )
 
-  val sourcecode = MillCommunityProject(
+  lazy val sourcecode = MillCommunityProject(
     project = "sourcecode",
     baseCommand = s"sourcecode.jvm[$compilerVersion]",
   )
 
-  val oslib = MillCommunityProject(
+  lazy val oslib = MillCommunityProject(
     project = "os-lib",
     baseCommand = s"os[$compilerVersion]",
     dependencies = List(utest, sourcecode)
   )
 
-  val oslibWatch = MillCommunityProject(
+  lazy val oslibWatch = MillCommunityProject(
     project = "os-lib",
     baseCommand = s"os.watch[$compilerVersion]",
     dependencies = List(utest, sourcecode)
   )
 
-  val intent = SbtCommunityProject(
+  lazy val ujson = MillCommunityProject(
+    project = "upickle",
+    baseCommand = s"ujson.jvm[$compilerVersion]",
+    dependencies = List(scalatest, scalacheck, scalatestplusScalacheck)
+  )
+
+  lazy val intent = SbtCommunityProject(
     project       = "intent",
     sbtTestCommand   = "test",
     sbtUpdateCommand = "update"
   )
 
-  val algebra = SbtCommunityProject(
+  lazy val algebra = SbtCommunityProject(
     project       = "algebra",
     sbtTestCommand   = "coreJVM/compile",
     sbtUpdateCommand = "coreJVM/update"
   )
 
-  val scalacheck = SbtCommunityProject(
+  lazy val scalacheck = SbtCommunityProject(
     project       = "scalacheck",
     sbtTestCommand   = "jvm/test",
-    sbtUpdateCommand = "jvm/test:update"
+    sbtUpdateCommand = "jvm/test:update",
+    sbtPublishCommand = ";set jvm/publishArtifact in (Compile, packageDoc) := false ;jvm/publishLocal"
   )
 
-  val scalatest = SbtCommunityProject(
+  lazy val scalatest = SbtCommunityProject(
     project       = "scalatest",
     sbtTestCommand   = ";scalacticDotty/clean;scalacticTestDotty/test;scalatestTestDotty/test",
-    sbtUpdateCommand = "scalatest/update"
+    sbtUpdateCommand = "scalatest/update",
+    sbtPublishCommand = ";scalacticDotty/publishLocal; scalatestDotty/publishLocal"
+  )
+
+  lazy val scalatestplusScalacheck = SbtCommunityProject(
+    project = "scalatestplus-scalacheck",
+    sbtTestCommand = "scalatestPlusScalaCheckJVM/compile",  // TODO: compile only because tests are prone to java.lang.OutOfMemoryError: Metaspace
+    sbtUpdateCommand = "scalatestPlusScalaCheckJVM/update",
+    sbtPublishCommand = "scalatestPlusScalaCheckJVM/publishLocal",
+    dependencies = List(scalatest, scalacheck)
```

Comment:
The dependency on `scalatest` and `scalacheck` is not effective, if the versions don't match in dependency specification.

Reply:
It should be fine FTTB. Assuming that specifying the correct dependency is the responsibility of the one adding a new project to the build, and that these custom dependencies are local, the test would just fail to resolve the dependencies if they are misspecified in any of the concerned projects.